pkg/deployment/cert.go

```diff
@@ -112,15 +111,18 @@ func EnsureTLSCerts(ctx context.Context, helper *helper.Helper,
 
 		if service.Spec.TLSCert.Issuer == "" {
 			// by default, use the internal root CA
-			// issuer = certmanager.RootCAIssuerInternalLabel
-			// TODO(alee) Temporarily set this to the issuer name
-			issuer = tls.DefaultCAPrefix + string(endpoint.EndpointInternal)
+			issuerLabelSelector = map[string]string{certmanager.RootCAIssuerInternalLabel: ""}
 		} else {
-			issuer = service.Spec.TLSCert.Issuer
+			issuerLabelSelector = map[string]string{service.Spec.TLSCert.Issuer: ""}
```

Comment:
should the issuer parameter in the service spec be the issuer name or the label?, e.g. https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/services/dataplane_v1beta1_openstackdataplaneservice_libvirt.yaml#L14C13-L14C26

I mean I am wondering if it should be something like this:

```
		if service.Spec.TLSCert.Issuer == "" {
			i, err := certmanager.GetIssuerByLabels(ctx, helper, "", issuerLabelSelector)
			if err != nil {
				return &result, err
			}
                        issuer = i.Name
		} else {
			issuer = service.Spec.TLSCert.Issuer
		}

		certSecret, result, err = GetTLSNodeCert(ctx, helper, instance, secretName,
			issuer, labels, hosts, ips, nil)
```

Reply:
Well, this depends on whether we plan to create labels for other "standard" CAs - like for example, for OVN or libvirt, and have those labels specified somewhere central like lib-common and used on the control plane.   If so, then we need to use labels.

Its relatively trivial to add a label to an issuer if someone wants to use a different issuer. So the use of names vs. labels is really a design/convenience choice.  Labels across the board would be consistent though.

@stuggi @olliewalsh  - so what do we plan to do for OVN etc, ?

Reply:
as discussed on slack, will use labels